### PR TITLE
Generate release notes for 7.0.0-alpha1.

### PR DESCRIPTION
**Pull request by jpountz**
We were missing release notes for 7.0.0-alpha1. I generated them by running
the release-notes script with a quick hack that filtered out pull requests
that had been closed on or after 2018-11-14.

Some breaking changes had been documented in the release notes rather than
the migration guide so I moved them.

### CODE REVIEW

File: docs/reference/release-notes/7.0.0-alpha1.asciidoc [hunk 1]
```diff
@@ -1,32 +1,2148 @@
 [[release-notes-7.0.0-alpha1]]
 == {es} version 7.0.0-alpha1
 
-The changes listed below have been released for the first time in Elasticsearch 7.0.0-alpha1.
-
```
[the remark this answers is not in the document]
Comment (jpountz) [reply docs/reference/release-notes/7.0.0-alpha1.asciidoc]: @lcawl I generated release notes with the script that is available at `dev-tools/es_release_notes.pl`. It doesn't include this sentence, plus this sentence would be incorrect given the current way that the script works due to the fact that it doesn't try to exclude changes that were also included in 6.x.

[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/release-notes/7.0.0-alpha1.asciidoc]: I think these release notes were likely modelled after https://www.elastic.co/guide/en/elasticsearch/reference/6.0/release-notes-6.0.0-alpha1.html
It's worth noting that the Elasticsearch Release Notes process is under review.  My hope is that it'll end up generating release notes that are better aligned with other products' with respect to what's included in the PR list.

[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/release-notes/7.0.0-alpha1.asciidoc]: Even better, if we end up with one process and one tool that can be used across all our products' release notes, it'd be great.

**Comment by jpountz** [comment]
Closing as this PR needs to apply to 7.x now. I'll reopen a new one.

**Comment by jpountz** [comment]
Reopened at #39899. I applied @lcawl's suggestion of only including changes that were specific to 7.0, which could be done by uncommenting one line in the script.

**Pull request closed without merging**